File: photo-cube-client-react/src/components/Middle/ThreeBrowser/CubeObject.ts
import ObjectTagRelation from "./ObjectTagRelation";

/**
 * PhotoCube Client repressentation of an Object in the M^3 datamodel.
 * Is very similar to CubeObject.cs in PhotoCube Server implementation.
 */
export default interface CubeObject{
    Id: number,
    FileName: string|null,
    FileType: number,
    PhotoId: number,
    Photo: any,
    ObjectTagRelations: ObjectTagRelation[] | null,
    ThumbnailId: number,
    Thumbnail: any
}

File: Server/ObjectCubeServer/ObjectCubeServer/Models/DomainClasses/CubeObject.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ObjectCubeServer.Models.DomainClasses
{
    /// <summary>
    /// Repressents an Object in the M^3 datamodel.
    /// Has a fileName and a Thumbnail (visual presentation of the object).
    /// Is tagged with tags in ObjectTagRelation (many to many relation table to Tag).
    /// In the fututre i can repressent different types of files.
    /// </summary>
    public class CubeObject
    {
        [Key]
        public int Id { get; set; }

        public string FileName { get; set; }

        public FileType FileType { get; set; }

        public int? PhotoId { get; set; }
        public Photo Photo { get; set; }
        
        public List<ObjectTagRelation> ObjectTagRelations { get; set; }

        public int ThumbnailId { get; set; }
        public Thumbnail Thumbnail { get; set; }
    }
}
